go/protocol/keybase1/extras.go

```diff
@@ -1619,25 +1619,20 @@ func (t TeamMembers) AllUIDs() []UID {
 	return all
 }
 
-func (t TeamMembers) AllUserVersions() []UserVersion {
-	m := make(map[UID]UserVersion)
+func (t TeamMembers) AllUserVersions() (res []UserVersion) {
 	for _, u := range t.Owners {
-		m[u.Uid] = u
+		res = append(res, u)
 	}
 	for _, u := range t.Admins {
-		m[u.Uid] = u
+		res = append(res, u)
 	}
 	for _, u := range t.Writers {
-		m[u.Uid] = u
+		res = append(res, u)
 	}
 	for _, u := range t.Readers {
-		m[u.Uid] = u
+		res = append(res, u)
 	}
-	var all []UserVersion
-	for _, uv := range m {
-		all = append(all, uv)
-	}
-	return all
+	return res
```

Comment:
What changed to make it ok to just append all of them as opposed to using the map to de-duplicate?  Or was the map never necessary?

Reply:
That's a good question, I was going to ask specifically about it - some code that I've written expected AllUserVersion to really return all UVs, even the outdated ones or duplicates for same UID. Worth noting is that server disallows adding multiple UVs for same UID now, adding a member of for specific has to remove old version of that person if there is one - see https://github.com/keybase/keybase/blob/master/helpers/team.iced#L834-L849

This function is only called in two places, one being list.go. That change also didn't break any tests.

cc @mlsteele

Reply:
Actually this is Max's function, can you take a look @maxtaco  ? Thanks

Reply:
will the same user be listed twice for a team in the UI? We've had that bug in the past

Reply:
There are multiple things to consider here:
1) team/list will return "duplicates" if there are multiple versions of a user in team
2) inactive users are marked with "(inactive because of a reset)" label in `team list-members` cli.
3) but UI does something to filter duplicates by username
4) also server does not allow you to add duplicates anymore, adding someone has to always remove previous version of that uid if present in the team

Since you mention this as a bug, so probably point 2 is a bug as well and I should change this.

Reply:
UI doesn't de-dupe anything, I have a team with this bug in it now.

Reply:
Is this on prod? Can you add me?